### PR TITLE
Feature #148 wind barbs

### PR DESCRIPTION
**Pull request by georgemccabe**
This PR adds support for rendering wind fields as vectors instead of contours.

Some notes that may be useful for future enhancements:

* Logic to determine how many vectors to display/skip uses hard-coded values and could be improved to be based on the projection/grid/etc.

* Some logic in vector_json.py is similar to geojson.py and could be refactored to clean up the code.

* Roughly 20 km spacing of wind vector points are stored in the json files. More wind vector points could potentially be added if performance isn't hindered too much.

A contour field for reference:
2m Temp Transferred/Size:
573 kB   / 2.15 MB
1.70 MB / 7.42 MB
1.73 MB / 7.62 MB

Sending all available points (before json format cleanup):
10m Wind (all points) Transferred/Size:
2 MB      / 26.48 MB
2.12 MB / 26.43 MB
2.10 MB / 26.43 MB

Using 20km spacing:
10m Wind Transferred/Size:
87.04 kB / 1.06 MB
88.89 kB / 1.06 MB
89.14 kB / 1.06 MB

Using 20 km spacing (after json format cleanup)
10m Wind - Transferred/Size:
86.39 kB / 940.89 kB
87.94 kB / 939.11 kB
87.77 kB / 939.01 kB

## Expected Differences ##

- [X] Do these changes modify the system output in any way? **[Yes]**</br>
Wind (10m and 3D) is now displayed as vectors. Previously wind direction and wind speed were contoured plots.

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Created output in a cluster and viewed TypeScript changes locally in a browser via angular to confirm wind fields are displayed as expected.

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review code changes. @fossell and @hahnd have already reviewed the display in a meeting

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

Updated list of graphics

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>

- [X] Please complete this pull request review by **3/20/2023**.</br>

## Pull Request Checklist ##
- [X] Review the source issue metadata (labels, project, and milestone).
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Project**
Select: **Milestone** as the version that will include these changes
Select: **Development** to link to the original development issue.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Comment by georgemccabe** [comment]
These resources may be helpful for improving the zoom logic and displaying the current scale:

https://openlayers.org/en/latest/examples/scale-line.html
https://gis.stackexchange.com/questions/242424/how-to-get-map-units-to-find-current-scale-in-openlayers

**Review by fossell — APPROVED** [review]
I'm reviewing and approving this PR based on meeting with @georgemccabe to display and demonstrate the new product locally. The new wind direction product looks great and performs as expected. Will plan to assess various new model domains after this feature is live to determine if further refinement is necessary. I have not inspected the actual code.

**Comment by hahnd** [comment]
I’m working my way through the code.  When I run the web app locally, the wind vectors do not display.  Are there specific cases I should look at?  I see the app making calls to the API to get the wind vectors and reasonable data in the responses.

**Comment by georgemccabe** [comment]
@hahnd, job [W64FF05E46D](https://app.wrfcloud.com/view/W64FF05E46D) was run on a cluster with my changes that are needed to view the wind products locally. I apologize for not including that info in the PR.

@fossell and I met this morning and decided to move forward with merging this PR so I can start the reinstall process with these changes. I will leave issue #148 open so you can provide feedback on these changes and we can make any adjustments that are needed.

File: python/src/wrfcloud/runtime/tools/vector_json.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        y_frac = y - y1
+
+        # compute lat/lon values with a linear interpolation
+        lat = lat1 + ((lat2 - lat1) * y_frac)
+        lon = lon1 + ((lon2 - lon1) * x_frac)
+
+        return round(lon, 5), round(lat, 5)
+
+
+def main():
```
Comment (hahnd) [inline python/src/wrfcloud/runtime/tools/vector_json.py:155]: We should add an entrypoint for this function in the setup.py to make it easier to call from the command line.

File: web/src/app/wrf-viewer/wrf-viewer.component.ts [hunk 2]
```diff
@@ -478,13 +563,56 @@ export class WrfViewerComponent implements OnInit
...
 
+    /**
+   * Set the style for vectors, e.g. wind
+   * @param feature
+   * @private
+   */
+  private selfVectorStyle(feature: any): Style[]
+  {
+    const vectorScale = this.getVectorScale(this.map?.getView().getZoom());
+    //console.log('scale: ' + vectorScale);
```
Comment (hahnd) [inline web/src/app/wrf-viewer/wrf-viewer.component.ts:581]: Delete this commented line.

File: web/src/app/wrf-viewer/wrf-viewer.component.ts [hunk 3]
```diff
@@ -496,6 +624,33 @@ export class WrfViewerComponent implements OnInit
...
+   * @param event
+   * @private
+   */
+
+  private doZoomChange(layer: WrfLayer, vectorLayer: VectorLayer<any>, event: any): void
+  {
+    const new_zoom = event.target.values_.zoom;
+    const new_spacing = this.getVectorSpacing(new_zoom);
+    const old_spacing = this.getVectorSpacing(layer.zoom);
+    //console.log('zoom: ' + layer.zoom + ' -> ' + new_zoom);
```
Comment (hahnd) [inline web/src/app/wrf-viewer/wrf-viewer.component.ts:641]: Delete this commented line.

File: python/src/wrfcloud/runtime/postproc.py [hunk 4]
```diff
@@ -217,34 +218,38 @@ def run(self) -> bool:
         """
         try:
             # create the geojson files
```
Comment (hahnd) [inline python/src/wrfcloud/runtime/postproc.py:220]: Looks like you're replacing all the geojson terminology here.  It still exists in a few comments.  We should also consider changing the API action name from `GetWrfGeoJson`.

**Review by hahnd — COMMENTED** [review]
Looks good.  A few minor change suggestions.

File: python/src/wrfcloud/runtime/postproc.py (hunk 4 above)
Comment (georgemccabe) [reply python/src/wrfcloud/runtime/postproc.py:220]: I created issue #168 to address this

**Pull request merged**